jkube-kit/jkube-kit-spring-boot/src/test/java/org/eclipse/jkube/springboot/SpringBootDevtoolsUtilsTest.java

```diff
@@ -61,17 +61,21 @@ void ensureSpringDevToolSecretToken_whenNoTokenFound_thenAppendTokenAndThrowExce
     boolean result = SpringBootDevtoolsUtils.ensureSpringDevToolSecretToken(javaProject);
 
     // Then
-    assertThat(result).isFalse();
-    Path srcApplicationProperties = temporaryFolder.toPath()
-        .resolve("src").resolve("main").resolve("resources").resolve("application.properties");
-    Path targetApplicationProperties = temporaryFolder.toPath()
-        .resolve("target").resolve("classes").resolve("application.properties");
-    assertThat(new String(Files.readAllBytes(srcApplicationProperties)))
-        .contains("# Remote secret added by jkube-kit-plugin")
-        .contains("spring.devtools.remote.secret=");
-    assertThat(new String(Files.readAllBytes(targetApplicationProperties)))
-        .contains("# Remote secret added by jkube-kit-plugin")
-        .contains("spring.devtools.remote.secret=");
+    thenSpringBootDevtoolsTokenAddedToProject(result);
+  }
+
+  @Test
+  void ensureSpringDevToolSecretToken_whenNoTokenFoundButTargetDirAlreadyExists_thenAppendTokenAndThrowException() throws IOException {
```

Comment:
What does `thenAppendTokenAndThrowException` mean in this context?
Not sure that an exception is thrown at any point

Reply:
During first execution when there is no devtools secret present, we add the token but fail the build by throwing exception

https://github.com/eclipse/jkube/blob/a8983aef7e35c0a3021d3a58c549cc9712ba1ae8/jkube-kit/jkube-kit-spring-boot/src/main/java/org/eclipse/jkube/springboot/generator/SpringBootGenerator.java#L73-L76

Reply:
Well, the test is not demonstrating that at all.

You want a test for the first clause and a test for the second.

Reply:
The logic for throwing exception if devtools secret wasn't found is part of SpringBootGenerator.

Second clause is covered by this test:

https://github.com/eclipse/jkube/blob/a8983aef7e35c0a3021d3a58c549cc9712ba1ae8/jkube-kit/jkube-kit-spring-boot/src/test/java/org/eclipse/jkube/springboot/generator/SpringBootGeneratorTest.java#L170-L184

Reply:
I'm not sure if I'm being clear.
The test description says that the method is throwing an exception `AndThrowException()`

I don't see anywhere in the test an assert exception was thrown.
Moreover, if this was actually the case, then we'd need to split the test in two.

Also, if by covered you mean code coverage, I really don't care it's covered elsewhere. We need to cover the expected behavior at whatever level we need to.


Again, to be clear, what I see very wrong here is the name of the test. It says the code is doing something that is not, or at least is not demonstrating it.

Reply:
You're right. I've updated test name.